fuzz_altsvc.cc

```diff
@@ -0,0 +1,83 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) 2017, Max Dymond, <cmeister2@gmail.com>, et al.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ ***************************************************************************/
+
+extern "C"
+{
+  #define HAVE_STRUCT_TIMEVAL // HACK to let it compile
+  #include <stdlib.h>
+  #include <signal.h>
+  #include <string.h>
+  #include <unistd.h>
+  #include <inttypes.h>
+  #include <curl/curl.h>
+  #include <assert.h>
+
+  enum alpnid {
+    ALPN_none = 0,
+    ALPN_h1 = CURLALTSVC_H1,
+    ALPN_h2 = CURLALTSVC_H2,
+    ALPN_h3 = CURLALTSVC_H3
+  };
+
+  struct altsvcinfo *Curl_altsvc_init(void);
+  CURLcode Curl_altsvc_parse(struct Curl_easy *data,
+                            struct altsvcinfo *altsvc, const char *value,
+                            enum alpnid srcalpn, const char *srchost,
+                            unsigned short srcport);
+  void Curl_altsvc_cleanup(struct altsvcinfo **altsvc);
+
+}
+
+#include <string>
+
+/* #define DEBUG(STMT)  STMT */
+#define DEBUG(STMT)
+
+
+/**
+ * Fuzzing entry point. This function is passed a buffer containing a test
+ * case.  This test case should drive the CURL fnmatch function.
+ */
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+  std::string s(reinterpret_cast<const char*>(data), size);
```

Comment:
Mildly worried that we could have a std::string buffer (which isn't null terminated) containing a buffer with a null in it, and what that would do for the fuzzing. Probably fine though.

(One of the things I've been wanting to get into the codebase is the [FuzzedDataProvider](https://github.com/google/fuzzing/blob/master/docs/split-inputs.md#fuzzed-data-provider) which might help here, but equally might just do the same thing under the covers)